File: app/src/main/java/com/yanxing/ui/UltraPtrFragment.java
package com.yanxing.ui;

import android.widget.ListView;

import com.yanxing.adapterlibrary.CommonAdapter;
import com.yanxing.adapterlibrary.ViewHolder;
import com.yanxing.base.BaseFragment;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import in.srain.cube.views.ptr.PtrClassicFrameLayout;
import in.srain.cube.views.ptr.PtrDefaultHandler;
import in.srain.cube.views.ptr.PtrFrameLayout;

/**
 * UltraPtr
 * Created by lishuangxiang on 2016/5/30.
 */
public class UltraPtrFragment extends BaseFragment {

    @BindView(R.id.listview)
    ListView mListView;

    @BindView(R.id.store_house_ptr_frame)
    PtrClassicFrameLayout mPtrClassicFrameLayout;

    private List<String> mList = new ArrayList<String>();
    private CommonAdapter<String> mCommonAdapter;

    @Override
    protected int getLayoutResID() {
        return R.layout.fragment_ultra_ptr;
    }

    @Override
    protected void afterInstanceView() {
        mPtrClassicFrameLayout.setPtrHandler(new PtrDefaultHandler() {
            @Override
            public void onRefreshBegin(PtrFrameLayout frame) {
                mPtrClassicFrameLayout.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        load();
                        mPtrClassicFrameLayout.refreshComplete();
                    }
                }, 1000);
            }
        });
        mPtrClassicFrameLayout.autoRefresh(true);
    }

    //javaScript
    public void load() {
        for (int i=0;i<10;i++){
            mList.add(String.valueOf(i));
        }
        mCommonAdapter = new CommonAdapter<String>(mList,R.layout.adapter_content) {
            @Override
            public void onBindViewHolder(ViewHolder viewHolder, int position) {
                viewHolder.setText(R.id.name,mDataList.get(position));
            }
        };
        mListView.setAdapter(mCommonAdapter);
    }
}

File: app/src/main/java/com/yanxing/ui/UltraPtrFragment.kt
package com.yanxing.ui


import com.yanxing.adapterlibrary.CommonAdapter
import com.yanxing.adapterlibrary.ViewHolder
import com.yanxing.base.BaseFragment

import java.util.ArrayList

import `in`.srain.cube.views.ptr.PtrDefaultHandler
import `in`.srain.cube.views.ptr.PtrFrameLayout
import kotlinx.android.synthetic.main.fragment_ultra_ptr.*

/**
 * UltraPtr
 * Created by lishuangxiang on 2016/5/30.
 */
class UltraPtrFragment : BaseFragment() {

    private val mList = ArrayList<String>()
    private var mCommonAdapter: CommonAdapter<String>? = null

    override fun getLayoutResID(): Int {
        return R.layout.fragment_ultra_ptr
    }

    override fun afterInstanceView() {
        ptrClassicFrameLayout.setPtrHandler(object : PtrDefaultHandler() {
            override fun onRefreshBegin(frame: PtrFrameLayout) {
                ptrClassicFrameLayout.postDelayed({
                    load()
                    ptrClassicFrameLayout.refreshComplete()
                }, 1000)
            }
        })
        ptrClassicFrameLayout.autoRefresh(true)
    }

    //javaScript
    fun load() {
        for (i in 0..9) {
            mList.add(i.toString())
        }
        mCommonAdapter = object : CommonAdapter<String>(mList, R.layout.adapter_content) {
            override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
                viewHolder.setText(R.id.name, mDataList[position])
            }
        }
        listview.adapter = mCommonAdapter
    }
}
